### PR TITLE
Update client and checkout

### PR DESCRIPTION
**Pull request by shiroginne**
Update Client and Checkout.
Add specs.
Fix bugs & small issues.

### CODE REVIEW

File: spec/be_gateway/checkout_spec.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          response = checkout.get_token(request_params)
+
+          expect(response.invalid?).to eq(true)
+          expect(response['errors']).not_to be_empty
+        end        
+      end
+    end
+
+  end
+end
```
Comment (MihailDV) [inline spec/be_gateway/checkout_spec.rb]: Could you add empty line?

File: lib/be_gateway/client.rb [hunk 2]
```diff
@@ -1,43 +1,13 @@
...
-    def chargeback(params)
-      response = post "/transactions/chargebacks", { request: params }
-      make_response(response)
-    end
-
-    def fraud_advice(params)
-      response = post "/transactions/fraud_advices", { request: params }
-      make_response(response)
+    TRANSACTIONS.each do |tr_type|
+      define_method tr_type do |params|
```
Comment (alovak) [inline lib/be_gateway/client.rb]: Interesting, that the first argument of define_method should be a symbol: http://ruby-doc.org/core-2.2.0/Module.html#method-i-define_method

how does it work here? :)

Comment (shiroginne) [inline lib/be_gateway/client.rb]: It's an interesting question! Maybe it's a ruby bug? But we have similar definition (string not symbil) in other gems, and it work well.

**Comment by shiroginne** [comment]
up! :)

**Comment by MihailDV** [comment]
:shipit:

**Comment by shiroginne** [comment]
bump! :coffee:

**Comment by seongreen** [comment]
:ok:

**Pull request merged**